src/terminal/char_grid/styles.rs

```diff
@@ -29,9 +29,25 @@ pub struct Styles {
     pub strikethrough: bool,
     pub inverted: bool,
     pub blink: bool,
+    config: Config,
```

Comment:
Similarly, I'd rather not duplicate the whole config struct for every `Styles` struct. Instead of keeping this around, pass the `Config` from the `CharGrid` as an argument to the `update` function.